### DISCUSSION TITLE
Distinguish first-time processing of repeatable jobs

### DISCUSSION THREAD

**Discussion by Thore1954** [taskforcesh/bullmq#1738] [Q&A] [answered]
I'm in love with the new bull since I migrated one project late last year. Now I am so reliant on it's extensive guides and clear implementation that I'm migrating others as well. Amazing "job handling"! (pun very much intended)

I have a requirement where I need to determine when the repeatable job was last run. I found out that the `timestamp` of the current job is an approximation of the `processedOn` of the last job except when there was no last job in which case it's the timestamp when the job was created with `Queue.add`.

Is there any way to distinguish between the first processing and later ones?

**Comment by manast** [comment]
Hmm, that would require something like finding the last job that was processed by the given repeatable "meta" job and then examining the ```finishedOn``` field. I am not sure this is easy to do without some new API 🤔

**Reply by Thore1954** [reply]
It wouldn't be an option if `removeOnComplete` is set. Is there another way to achieve this by e.g. passing certain args to `Queue.add`?

**Reply by manast** [reply]
Sorry, I do not understand your question.

**Comment by Thore1954** [comment, marked as answer]
For anybody else looking for this, I found out that the `count` property in `RepeatOptions` does exactly this. It is set to 1 for the first job and keeps incrementing for later ones. There's just a weird bug where it would reset to 1 every time the app is shut down for long but if your app doesn't shut down then it's fine.

**Reply by manast** [reply]
It is possible this reset of the count field happens if you add the same repeatable job once more time.
